src/handlers/gestures/gestureBuilder.ts

```diff
@@ -0,0 +1,113 @@
+import { SimpleGesture } from './simpleGestures';
+
+enum Relation {
+  Simultaneous,
+  Exclusive,
+  RequireToFail,
+}
+
+type PendingGesture = {
+  relation: Relation;
+  gesture: SimpleGesture;
+};
+
+export class GestureBuilder {
+  private pendingGestures: PendingGesture[] = [];
+
+  constructor(base: SimpleGesture) {
+    this.addGesture({ relation: Relation.Exclusive, gesture: base });
+  }
+
+  private addGesture(gesture: PendingGesture): GestureBuilder {
+    this.pendingGestures.push(gesture);
+    return this;
+  }
+
+  simultaneousWith(gesture: SimpleGesture): GestureBuilder {
+    return this.addGesture({
+      relation: Relation.Simultaneous,
+      gesture: gesture,
+    });
+  }
+
+  exclusiveWith(gesture: SimpleGesture): GestureBuilder {
+    return this.addGesture({
+      relation: Relation.Exclusive,
+      gesture: gesture,
+    });
+  }
+
+  requireToFail(gesture: SimpleGesture): GestureBuilder {
+    this.pendingGestures.push({
+      relation: Relation.RequireToFail,
+      gesture: gesture,
+    });
+
+    return this;
+  }
+
+  build(): BuiltGesture {
+    const result = new BuiltGesture(this.prepare);
+    result.gestures = this.pendingGestures.map((pending) => pending.gesture);
+    return result;
+  }
+
+  prepare = () => {
+    const simultaneous: number[] = [];
+    const waitFor: number[] = [];
+
+    for (let i = this.pendingGestures.length - 1; i >= 0; i--) {
+      const pendingGesture = this.pendingGestures[i];
+      pendingGesture.gesture.prepare();
+
+      const newConfig = { ...pendingGesture.gesture.config };
+
+      if (newConfig.simultaneousWith) {
+        newConfig.simultaneousWith = [
+          ...newConfig.simultaneousWith,
+          ...simultaneous,
+        ];
+      } else {
+        newConfig.simultaneousWith = [...simultaneous];
+      }
```

Comment:
We could add some small helper, like `overrideConfig(oldConfig, newConfig)` and use it:
```ts
newConfig.simultaneousWith = overrideConfig(newConfig.simultaneousWith, simultaneous)
```

`overrideConfig` maybe isn't the best name for it 🤷

Reply:
Added in f181ba99141486b3c88755a4475c669ad0b6d3e5, is `extendRelation` name ok?